crates/pathfinder/src/state/block_hash.rs

```diff
@@ -38,27 +44,26 @@ pub enum VerifyResult {
 ///
 /// See the `compute_block_hash.py` helper script that uses the cairo-lang
 /// Python implementation to compute the block hash for details.
-pub fn verify_block_hash(
+pub fn verify_gateway_block_hash(
     block: &Block,
     chain: Chain,
     chain_id: ChainId,
-    expected_block_hash: BlockHash,
 ) -> Result<VerifyResult> {
-    let meta_info = meta::for_chain(chain);
-    if !meta_info.can_verify(block.block_number) {
-        return Ok(VerifyResult::NotVerifiable);
-    }
-
-    let num_transactions: u64 = block
-        .transactions
-        .len()
-        .try_into()
-        .expect("too many transactions in block");
-
     let transaction_final_hash_type =
         TransactionCommitmentFinalHashType::for_version(&block.starknet_version);
     let transaction_commitment =
         calculate_transaction_commitment(&block.transactions, transaction_final_hash_type)?;
+
+    let mut block_header_data = BlockHeaderVerificationData::from(block);
+
+    // Older blocks on mainnet don't carry a precalculated transaction
+    // commitment.
```

Comment:
Well that's frustrating.. we should ask Starkware to clean this up? Surely something they can backfill. Along with the event commitment..